provider/blob_online.go

```diff
@@ -0,0 +1,204 @@
+package provider
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+)
+
+type OnlineBlobConfig struct {
+	Type   BlobStoreType
+	Config BlobStoreConfig
+}
+
+func (online OnlineBlobConfig) Serialized() SerializedConfig {
+	config, err := json.Marshal(online)
+	if err != nil {
+		panic(err)
+	}
+	return config
+}
+
+func (online *OnlineBlobConfig) Deserialize(config SerializedConfig) error {
+	err := json.Unmarshal(config, online)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+type OnlineBlobStore struct {
+	BlobStore
+	BaseProvider
+}
+
+func blobOnlineStoreFactory(serialized SerializedConfig) (Provider, error) {
+	onlineBlobConfig := &OnlineBlobConfig{}
+	if err := onlineBlobConfig.Deserialize(serialized); err != nil {
+		return nil, err
+	}
+	return NewOnlineBlobStore(onlineBlobConfig)
+}
+
+func NewOnlineBlobStore(config *OnlineBlobConfig) (*OnlineBlobStore, error) {
+	blobStore, err := CreateBlobStore(string(config.Type), Config(config.Config))
+	if err != nil {
+		return nil, fmt.Errorf("could not create blob store: %v", err)
+	}
+	return &OnlineBlobStore{
+		blobStore,
+		BaseProvider{
+			ProviderType:   BlobOnline,
+			ProviderConfig: config.Serialized(),
+		},
+	}, nil
+}
+
+func (store *OnlineBlobStore) AsOnlineStore() (OnlineStore, error) {
+	return store, nil
+}
+
+func blobTableKey(feature, variant string) string {
+	return fmt.Sprintf("tables/%s/%s", feature, variant)
```

Comment:
Are normal files and featureform files differentiated with this key? Just so people know which ones we own

Reply:
yeah i should prepend .featureform to both this and the entity keys.